composeApp/src/commonMain/kotlin/sp/bvantur/tasky/event/presentation/CreateEventViewModel.kt

```diff
@@ -0,0 +1,70 @@
+package sp.bvantur.tasky.event.presentation
+
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.launch
+import sp.bvantur.tasky.core.domain.DispatcherProvider
+import sp.bvantur.tasky.core.presentation.SingleEventHandler
+import sp.bvantur.tasky.core.presentation.SingleEventHandlerImpl
+import sp.bvantur.tasky.core.presentation.TextData
+import sp.bvantur.tasky.core.presentation.ViewModelUserActionHandler
+import sp.bvantur.tasky.core.presentation.ViewStateViewModel
+import sp.bvantur.tasky.event.ui.model.CreateEventModel
+import sp.bvantur.tasky.event.ui.model.SingleInputModel
+
+class CreateEventViewModel(dispatcherProvider: DispatcherProvider) :
+    ViewStateViewModel<CreateEventViewState>(
+        initialViewState = CreateEventViewState(),
+        dispatcherProvider = dispatcherProvider
+    ),
+    ViewModelUserActionHandler<CreateEventUserAction>,
+    SingleEventHandler<CreateEventSingleEvent> by SingleEventHandlerImpl(dispatcherProvider) {
+
+    fun onLoadInitialData(eventData: CreateEventModel) {
+        val title = eventData.title?.let {
+            TextData.DynamicString(it)
+        } ?: viewStateFlow.value.title
+
+        val description = eventData.description?.let {
+            TextData.DynamicString(it)
+        } ?: viewStateFlow.value.description
+
+        viewModelScope.launch {
+            emitViewState { viewState ->
+                viewState.copy(title = title, description = description)
+            }
```

Comment:
Could it make sense to inline the lambda from that function, so you don't always need to call it in a coroutine scope when not needing that?

Reply:
Ah I forgot you had this `withContext` in there 😅 I'm still not a fan of this function, it adds so much boilerplate by always having to update state in a coroutine scope. How often do you really need to explicitly switch to the Main dispatcher to update state? 99% of ViewModel code is usually executed on the main thread anyways and even if you update the state from another thread, what could go wrong with that? The `update` will already make sure there are no race conditions.
Maybe your thinking is that UI should be updated on the main thread only, but this code is just about updating the state, or in other words, assigning a different value to a variable. The actual UI update happens in your composable and will certainly be done on the main thread, regardless of where the state was updated.

Reply:
Good point. I will fix that. Thank you. :)